compiler/rustc_session/src/config.rs

```diff
@@ -1890,6 +1873,15 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
 
     let pretty = parse_pretty(matches, &debugging_opts, error_format);
 
+    if !debugging_opts.unstable_options
+        && !target_triple.triple().contains("darwin")
```

Comment:
Are custom target specs stable? Combined with `-Zbuild-std` once it becomes stable would allow faking macOS for thevpurpose of this check, while actually building for another target.

Reply:
Do you have a proposed alternative for how to change this?

Reply:
Maybe the llvm target could be checked instead? That will ensure that at the very least the object file format is Mach-O, for which macOS is pretty much the only OS that supports it.

Reply:
Is this really worth pursuing at this point? Everything about custom target specs is pseudo-unstable anyway and I'm not sure if this information is otherwise trivially accessible at this point. This is just a temporary workaround anyway until split-debuginfo is implemented and vetted on other platforms.

Reply:
I am fine with leaving this as is then.